extension/json/include/json_scan.hpp

```diff
@@ -0,0 +1,194 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// json_scan.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "buffered_json_reader.hpp"
+#include "duckdb/common/mutex.hpp"
+#include "duckdb/function/table_function.hpp"
+
+namespace duckdb {
+
+struct JSONScanLocalState;
+
+struct JSONScanData : public TableFunctionData {
```

Comment:
Does this all need to be in the header, or could it be moved to the `.cpp` file?

Reply:
The idea for this is that we have a shared `json_scan.hpp` file for all the future JSON table functions. In this case, I've re-implemented `read_(nd)json_objects`, which uses these common functionality from `json_scan.hpp`. In the future, we'll add more JSON table functions, which need a lot of the same functionality too. So I figured it was best to add it to the `.hpp` file.

Reply:
Fair enough